nectar/src/command/trade.rs

```diff
@@ -239,40 +238,60 @@ fn init_dai_balance_updates(
     (future, receiver)
 }
 
-// TODO: Move this to a struct that owns the channel
-#[allow(clippy::too_many_arguments)]
-async fn execute_swap(
-    db: Arc<Database>,
-    bitcoin_wallet: Arc<bitcoin::Wallet>,
-    ethereum_wallet: Arc<ethereum::Wallet>,
-    bitcoin_connector: Arc<comit::btsieve::bitcoin::BitcoindConnector>,
-    ethereum_connector: Arc<comit::btsieve::ethereum::Web3Connector>,
-    mut finished_swap_sender: Sender<FinishedSwap>,
-    swap: SwapKind,
-) -> anyhow::Result<()> {
-    db.insert_swap(swap.clone()).await?;
+#[derive(Debug, Clone)]
+struct SwapExecutor {
+    finished_swap_sender: Sender<FinishedSwap>,
+}
 
-    swap.execute(
-        db,
-        bitcoin_wallet,
-        ethereum_wallet,
-        bitcoin_connector,
-        ethereum_connector,
-    )
-    .await?;
-
-    let _ = finished_swap_sender
-        .send(FinishedSwap::new(
-            swap.clone(),
-            swap.params().taker,
-            chrono::Utc::now(),
-        ))
-        .await
-        .map_err(|_| {
-            tracing::trace!("Error when sending execution finished from sender to receiver.")
-        });
+impl SwapExecutor {
+    pub fn new() -> (Self, Receiver<FinishedSwap>) {
+        let (finished_swap_sender, finished_swap_receiver) =
+            futures::channel::mpsc::channel::<FinishedSwap>(ENSURED_CONSUME_ZERO_BUFFER);
 
-    Ok(())
+        (
+            Self {
+                finished_swap_sender,
+            },
+            finished_swap_receiver,
+        )
+    }
+}
+
+impl SwapExecutor {
+    async fn run(
+        mut self,
+        db: Arc<Database>,
+        bitcoin_wallet: Arc<bitcoin::Wallet>,
+        ethereum_wallet: Arc<ethereum::Wallet>,
+        bitcoin_connector: Arc<comit::btsieve::bitcoin::BitcoindConnector>,
+        ethereum_connector: Arc<comit::btsieve::ethereum::Web3Connector>,
```

Comment:
Could the executor possibly own these?

Reply:
yes I moved the connectors, I guess I did the commit after your review.